File: management/src/main/kotlin/services/ClassLevelService.kt
package com.education.services

import com.education.models.ClassLevelDAO
import com.education.models.ClassLevelDTO
import com.education.models.CreateClassLevelRequest
import com.education.models.UpdateClassLevelRequest
import com.education.repositories.ClassLevelRepository

interface ClassLevelService {
    suspend fun getAllClassLevels(organizationId: Int): List<ClassLevelDTO>
    suspend fun getClassLevelById(classLevelId: Int): ClassLevelDAO?
    suspend fun createClassLevel(classLevelRequest: CreateClassLevelRequest): Int
    suspend fun updateClassLevel(classLevelId: Int, updateClassLevelRequest: UpdateClassLevelRequest): Boolean
    suspend fun deleteClassLevel(classLevelId: Int): Boolean
}

class ClassLevelServiceImpl(
    private val classLevelRepository: ClassLevelRepository,
) : ClassLevelService {

    override suspend fun getAllClassLevels(organizationId: Int): List<ClassLevelDTO> {
        return classLevelRepository.getAllClassLevels(organizationId)
    }

    override suspend fun getClassLevelById(classLevelId: Int): ClassLevelDAO? {
        return classLevelRepository.getClassLevelById(classLevelId)
    }

    override suspend fun createClassLevel(classLevelRequest: CreateClassLevelRequest): Int {
        return classLevelRepository.createClassLevel(classLevelRequest)
    }

    override suspend fun updateClassLevel(classLevelId: Int, updateClassLevelRequest: UpdateClassLevelRequest): Boolean {
        return classLevelRepository.updateClassLevel(classLevelId, updateClassLevelRequest)
    }

    override suspend fun deleteClassLevel(classLevelId: Int): Boolean {
        return classLevelRepository.deleteClassLevel(classLevelId)
    }
} 

File: management/management-clientside/src/services/classLevelService.ts
import { 
    ClassLevelApi, 
    ComEducationModelsCreateClassLevelRequest, 
    ComEducationModelsUpdateClassLevelRequest,
    ComEducationModelsClassLevelDTO,
    ComEducationModelsClassLevelDAO
} from '../generated/api';
import { createApiConfig } from './generated-api';

// Create Class Level API instance
const classLevelApi = new ClassLevelApi(createApiConfig());

export async function getClassLevels(orgId: number) {
    try {
        return await classLevelApi.getClassLevels({ orgId });
    } catch (error) {
        console.error('Error fetching class levels:', error);
        throw error;
    }
}

export async function getClassLevelById(orgId: number, classLevelId: number) {
    try {
        return await classLevelApi.getClassLevelById({ orgId, classLevelId });
    } catch (error) {
        console.error('Error fetching class level by ID:', error);
        throw error;
    }
}

export async function createClassLevel(orgId: number, classLevelData: Omit<ComEducationModelsCreateClassLevelRequest, 'organizationId'>) {
    try {
        const request = {
            comEducationModelsCreateClassLevelRequest: {
                ...classLevelData,
                organizationId: orgId
            },
            orgId: orgId
        };
        return await classLevelApi.createClassLevel(request);
    } catch (error) {
        console.error('Error creating class level:', error);
        throw error;
    }
}

export async function updateClassLevel(orgId: number, classLevelId: number, classLevelData: Omit<ComEducationModelsUpdateClassLevelRequest, 'organizationId'>) {
    try {
        const request = {
            comEducationModelsUpdateClassLevelRequest: {
                ...classLevelData,
                organizationId: orgId
            },
            orgId: orgId,
            classLevelId: classLevelId
        };
        return await classLevelApi.updateClassLevel(request);
    } catch (error) {
        console.error('Error updating class level:', error);
        throw error;
    }
}

export async function deleteClassLevel(orgId: number, classLevelId: number) {
    try {
        return await classLevelApi.deleteClassLevel({ orgId, classLevelId });
    } catch (error) {
        console.error('Error deleting class level:', error);
        throw error;
    }
}

// Export types for easier usage
export type { 
    ComEducationModelsCreateClassLevelRequest as CreateClassLevelRequest,
    ComEducationModelsUpdateClassLevelRequest as UpdateClassLevelRequest,
    ComEducationModelsClassLevelDTO as ClassLevelDTO,
    ComEducationModelsClassLevelDAO as ClassLevelDAO
}; 